pkgs/build-support/docker/default.nix

```diff
@@ -82,7 +82,7 @@ rec {
     export PATH=${shadow}/bin:$PATH
     mkdir -p /etc/pam.d
     if [[ ! -f /etc/passwd ]]; then
-      echo "root:x:0:0::/root:/bin/sh" > /etc/passwd
+      echo "root:x:0:0::/root:${stdenv.shell}" > /etc/passwd
```

Comment:
IIRC you can only use paths from `/etc/shells` as a login shell.  Universal paths like `/run/current-system/...` seem less likely to break that.